bpf/types.h

```diff
@@ -138,24 +149,21 @@ typedef struct flow_id_t {
     u8 icmp_code;
     // OS interface index
     u32 if_index;
-} __attribute__((packed)) flow_id;
+} flow_id;
```

Comment:
without packed the generated golang structure will have padding and will be hard to predict the size of the struct which might fail unit test

Reply:
Removing packed was one of the things that really improved performances. And also the spin lock can't be in a packed struct (or at least, linter complains, I guess for good reasons?). So having a unit-test harder to maintain is IMO a lesser concern? (btw I want to remove that test but I saw you're commenting on that)

Reply:
do u recall what was the linter error ? plus not very clear on how removing pack would help performance that much tbh, I think generating predicitable userspace struct is important but let see the linter error 1st

Reply:
(I don't have in mind what the error was, but definitely was something wrong with lock+packed)

Reply:
I found this https://github.com/cilium/ebpf/blob/f0eec7efba9dd069f4d937d3096e276c7911a1c0/memory.go#L26 that is very likely explains the userspace cpu improvements